src/api/v2/quickPickWizard/QuickPickAppResourceStep.ts

```diff
@@ -3,20 +3,34 @@
  *  Licensed under the MIT License. See License.txt in the project root for license information.
  *--------------------------------------------------------------------------------------------*/
 
-import { AzureWizardPromptStep, IAzureQuickPickItem, IWizardOptions } from '@microsoft/vscode-azext-utils';
-import { PickAppResourceOptions } from '@microsoft/vscode-azext-utils/hostapi';
+import { AzureWizardPromptStep, IAzureQuickPickItem, IAzureQuickPickOptions, IWizardOptions } from '@microsoft/vscode-azext-utils';
+import { AppResource } from '@microsoft/vscode-azext-utils/hostapi';
 import { BranchDataProviderManager } from '../../../tree/v2/providers/BranchDataProviderManager';
 import { ApplicationResourceProviderManager } from '../providers/ApplicationResourceProviderManager';
-import { ApplicationResource, ResourceModelBase } from '../v2AzureResourcesApi';
-import { matchesContextValueFilter } from './ContextValueFilter';
+import { ApplicationResource, Filter, ResourceModelBase } from '../v2AzureResourcesApi';
 import { QuickPickAppResourceWizardContext } from './QuickPickAppResourceWizardContext';
 import { RecursiveQuickPickStep } from './RecursiveQuickPickStep';
 
+export interface PickAppResourceOptions2 extends IAzureQuickPickOptions {
+    /**
+     * Set this to pick a child of the selected app resource
+     */
+    childFilter?: Filter<ResourceModelBase>;
+
+    /**
+     * Whether `AppResourceTreeItem`s should be resolved before displaying them as quick picks, or only once one has been selected
+     * If a client extension needs to change label/description/something visible on the quick pick via `resolve`, set to true,
+     * otherwise set to false. Default will be false.
+     */
+    resolveQuickPicksBeforeDisplay?: boolean;
+}
+
 export class QuickPickAppResourceStep<TModel extends ResourceModelBase> extends AzureWizardPromptStep<QuickPickAppResourceWizardContext<TModel>> {
     public constructor(
         private readonly resourceProviderManager: ApplicationResourceProviderManager,
         private readonly branchDataProviderManager: BranchDataProviderManager,
-        private readonly options: PickAppResourceOptions
+        private readonly filter?: Filter<AppResource> | Filter<AppResource>[],

```

Comment:
I promoted `filter` from inside `options` to a top level parameter because in most cases we will want to filter the app resources.